### PR TITLE
Added support for sarif reports

### PR DESCRIPTION
**Pull request by armadi1809**
Still need to add tests but keeping this as draft in case someone notices something wrong. This should be ready soon.

### CODE REVIEW

**Comment by kehoecj** [comment]
@armadi1809 Looking good so far - just needs some tests!

**Comment by kehoecj** [comment]
> @armadi1809 Looking good so far - just needs some tests!

@armadi1809 Looks like this fork is getting a bit out of date. Please resolve the merge conflicts

**Comment by ccoVeille** [comment]
As usual, I'm likely to drop into the discussion, but it's the second time in a week I read about serif, and I have no idea what it could be.

The other project were I saw it was golangci-lint I think.

Could anyone explain me what it is, what it would allow to this project?

Thanks, Christophe

**Comment by kehoecj** [comment]
> As usual, I'm likely to drop into the discussion, but it's the second time in a week I read about serif, and I have no idea what it could be.
> 
> The other project were I saw it was golangci-lint I think.
> 
> Could anyone explain me what it is, what it would allow to this project?
> 
> Thanks, Christophe

>The Static Analysis Results Interchange Format (SARIF) is an industry standard format for the output of static analysis tools.

It started as a standard format for outputting static analysis results but now its use has expanded to general test output like JUnit.

**Comment by ccoVeille** [comment]
OK for thanks for replying me. I might have a look at it

File: cmd/validator/validator.go [hunk 1]
```diff
@@ -168,7 +168,9 @@ func getReporter(reportType, outputDest *string) reporter.Reporter {
 	case "junit":
 		return reporter.NewJunitReporter(*outputDest)
 	case "json":
-		return reporter.NewJSONReporter(*outputDest)
+		return reporter.NewJsonReporter(*outputDest)
```
Comment (kehoecj) [inline cmd/validator/validator.go:171]:
```suggestion
		return reporter.NewJSONReporter(*outputDest)
```

File: pkg/reporter/sarif_reporter.go [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+func (sr SarifReporter) Print(reports []Report) error {
+	report := createSarifReport(reports)
+
+	jsonBytes, err := json.MarshalIndent(report, "", "  ")
+	if err != nil {
+		return err
+	}
+
+	jsonBytes = append(jsonBytes, '\n')
+	fmt.Print(string(jsonBytes))
```
Comment (ccoVeille) [inline pkg/reporter/sarif_reporter.go:68]: This line looks like a left behind debug used for dev purpose

**Comment by kehoecj** [comment]
Work was never completed and #166 replaces it

**Pull request closed without merging**